### PR TITLE
Live tests

### PR DESCRIPTION
**Pull request by g2vinay**
Contains changes for Azure Key Vault Live Tests.

### CODE REVIEW

File: keyvault/data-plane/azure-keyvault-extensions/src/test/java/com/microsoft/azure/keyvault/extensions/test/KeyVaultKeyResolverDefaultProviderTest.java [hunk 1]
```diff
@@ -181,7 +181,7 @@ public void KeyVault_KeyVaultKeyResolver_Secret192Base64_KeyVaultKeyResolverDefa
          byte[] CEK      = { 0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77, (byte) 0x88, (byte) 0x99, (byte) 0xAA, (byte) 0xBB, (byte) 0xCC, (byte) 0xDD, (byte) 0xEE, (byte) 0xFF };
          byte[] EK       = { (byte) 0x96, 0x77, (byte) 0x8B, 0x25, (byte) 0xAE, 0x6C, (byte) 0xA4, 0x35, (byte) 0xF9, 0x2B, 0x5B, (byte) 0x97, (byte) 0xC0, 0x50, (byte) 0xAE, (byte) 0xD2, 0x46, (byte) 0x8A, (byte) 0xB8, (byte) 0xA1, 0x7A, (byte) 0xD8, 0x4E, 0x5D };
 
-         String TEST_SECRET_NAME = SECRET_NAME + "1";
+         String TEST_SECRET_NAME = SECRET_NAME + "4";
```
Comment (JonathanGiles) [inline keyvault/data-plane/azure-keyvault-extensions/src/test/java/com/microsoft/azure/keyvault/extensions/test/KeyVaultKeyResolverDefaultProviderTest.java]: It always makes me a little nervous when I see magic numbers being changed in unit tests. Why is this required?

File: template/client/src/main/java/com/microsoft/azure/template/Hello.java [hunk 2]
```diff
@@ -5,8 +5,15 @@
...
 public class Hello {
-    public String getMessage() {
+    /**
+     * Sample method.
+     * @return message
+     */
+    public final String getMessage() {
         return "hello";
     }
 }
```
Comment (JonathanGiles) [inline template/client/src/main/java/com/microsoft/azure/template/Hello.java]: This, and the package-info.java below, are unrelated changes, aren't they?

File: .azure-pipelines/client.yml [hunk 3]
```diff
@@ -2,32 +2,20 @@ trigger:
 - master
 
 jobs:
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Looks like you have some merge conflicts to handle which should get this file updated to what is currently committed in the repo. As it stands you are reverting some changes that were committed because your branch is behind master.

File: template/client/src/main/java/com/microsoft/azure/template/Hello.java (hunk 2 above)
Comment (lenala) [reply template/client/src/main/java/com/microsoft/azure/template/Hello.java]: They both caused checkstyle and because of that the build fail.

File: .azure-pipelines/client.test.live.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+- repo: self
+queue:
+  name: Hosted VS2017
+steps:
+- bash: |
+   LOG_PARAMS='-Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+   mvn test  -Dsurefire.rerunFailingTestsCount=3 $LOG_PARAMS -Dparallel=classes -DthreadCount=3 -DforkCount=1C -f pom.client.build.xml
+  displayName: 'Bash Script'
+  env:
+    ARM_CLIENTID: $(ARM_CLIENTID)
```
[the remark this answers is not in the document]
Comment (g2vinay) [reply .azure-pipelines/client.test.live.yml]: There are more Variables defined in the variables tab of dev ops pipeline.

File: .azure-pipelines/client.yml (hunk 3 above)
Comment (g2vinay) [reply .azure-pipelines/client.yml]: Conflicts resolved

File: .azure-pipelines/client.test.live.yml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+variables:
+  - group: KeyVault Test Secrets
+  - name: ARM_CLIENTID
+    value: $(java-keyvault-test-arm-client-id)
+  - name: ARM_CLIENTKEY
+    value: $(java-keyvault-test-arm-client-key)
+
+resources:
```
Comment (weshaggard) [inline .azure-pipelines/client.test.live.yml]: Why do you need "resources: self"?

File: .azure-pipelines/client.test.live.yml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+- repo: self
+queue:
+  name: Hosted VS2017
+steps:
+- bash: |
+   LOG_PARAMS='-Dorg.slf4j.simpleLogger.defaultLogLevel=error -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+   mvn test -Dhttp.keepAlive=false -Dsurefire.rerunFailingTestsCount=3 $LOG_PARAMS -f pom.client.build.xml 
+  displayName: 'Bash Script'
```
Comment (weshaggard) [inline .azure-pipelines/client.test.live.yml]: Please update this to represent what it is doing, like "Running Live Tests"

File: .azure-pipelines/client.test.live.yml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+    value: $(java-keyvault-test-arm-client-id)
+  - name: ARM_CLIENTKEY
+    value: $(java-keyvault-test-arm-client-key)
+
+resources:
+- repo: self
+queue:
+  name: Hosted VS2017
+steps:
+- bash: |
```
Comment (weshaggard) [inline .azure-pipelines/client.test.live.yml]: Is there a reason you aren't using the Maven@3 task to run the tests similar to https://github.com/Azure/azure-sdk-for-java/blob/master/.azure-pipelines/client.test.yml#L17?

In general we should try to do that to help be consistent and also that allows us to capture the test results into the test tab of the build which gives us a good view of the tests than ran and a history of them.

Comment (g2vinay) [reply .azure-pipelines/client.test.live.yml]: The environment were not read during the build when tried to configure as Maven task.

File: .azure-pipelines/client.test.live.yml (hunk 6 above)
Comment (g2vinay) [reply .azure-pipelines/client.test.live.yml]: Updated it.

File: .azure-pipelines/client.test.live.yml (hunk 5 above)
Comment (g2vinay) [reply .azure-pipelines/client.test.live.yml]: Not needed actually, 
removed.

File: keyvault/data-plane/pom.xml [hunk 8]
```diff
@@ -293,13 +293,13 @@
             <configuration>
                 <properties>
                     <property>
-<name>KEYVAULT_VAULTURI</name>
-       		<value>https://tifchen-keyvault-fancy.vault.azure.net</value>
+                          <name>KEYVAULT_VAULTURI</name>
+                          <value>https://azure-keyvault-3.vault.azure.net</value>
```
Comment (weshaggard) [inline keyvault/data-plane/pom.xml]: Looks like you are setting these in 3 places. One is here, one is below and one in the build definition. Do they need to be set in all these places?

Comment (g2vinay) [reply keyvault/data-plane/pom.xml]: Updated, the values should only be in environment variables section of pom.
Removed from other places.

File: keyvault/data-plane/pom.xml [hunk 9]
```diff
@@ -325,8 +312,8 @@
             </includes>
             <environmentVariables>
               <test.mode>${testMode}</test.mode>
-              <KEYVAULT_VAULTURI>https://tifchen-keyvault-fancy.vault.azure.net</KEYVAULT_VAULTURI>
-              <KEYVAULT_VAULTURI_ALT>https://tifchen-keyvault-fancier.vault.azure.net</KEYVAULT_VAULTURI_ALT>
+              <KEYVAULT_VAULTURI>https://azure-keyvault-3.vault.azure.net</KEYVAULT_VAULTURI>
```
Comment (weshaggard) [inline keyvault/data-plane/pom.xml]: What about here? Should we also remove these?

Comment (weshaggard) [reply keyvault/data-plane/pom.xml]: I see you removed them from the YAML instead so there is no duplication.
